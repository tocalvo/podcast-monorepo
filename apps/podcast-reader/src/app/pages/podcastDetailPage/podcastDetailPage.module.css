.container {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  margin: 20px 10px;
  justify-content: space-evenly;
  align-items: flex-start;
  gap: 15px;
}
