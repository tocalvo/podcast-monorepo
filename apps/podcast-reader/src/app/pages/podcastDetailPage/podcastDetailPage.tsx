import {
  EpisodesTable,
  PodcastBar,
  useItunesPodcastDetail,
  useItunesPodcasts,
} from '@org/podcasts';
import styles from './podcastDetailPage.module.css';

import { useParams } from 'react-router';

export const PodcastDetailPage: React.FC = () => {
  const { podcastId } = useParams();

  const { getPodcastResumeById } = useItunesPodcasts();
  const itunesPodcastResume = getPodcastResumeById(podcastId);

  const { episodes } = useItunesPodcastDetail(podcastId);

  if (!podcastId || !itunesPodcastResume) {
    return; // redirect to home or to 404
  }

  return (
    <div className={styles['container']}>
      <PodcastBar podcast={itunesPodcastResume} />
      <EpisodesTable podcastId={podcastId} episodes={episodes} />
    </div>
  );
};
export default PodcastDetailPage;
