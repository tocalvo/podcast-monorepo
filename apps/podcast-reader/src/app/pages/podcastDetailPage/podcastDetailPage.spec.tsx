import {
  ITunesPodcastKind,
  podcastDetailMock,
  podcastListMock,
} from '@org/podcasts';
import { render, screen } from '@testing-library/react';
import { PodcastDetailPage } from './podcastDetailPage';

import { BrowserRouter } from 'react-router';

vi.mock('@org/podcasts', async () => ({
  ...(await vi.importActual('@org/podcasts')),
  useItunesPodcasts: vi.fn(() => ({
    podcasts: podcastListMock.feed.entry,
    getPodcastResumeById: vi
      .fn()
      .mockReturnValue(podcastListMock.feed.entry[0]),
  })),
  useItunesPodcastDetail: vi.fn(() => ({
    podcastDetail: podcastDetailMock,
    episodes: podcastDetailMock.results.filter(
      (podcast) => podcast.kind === ITunesPodcastKind['podcast-episode']
    ),
    findEpisodeById: vi.fn().mockReturnValue(podcastDetailMock.results[1]),
  })),
}));

vi.mock('react-router', async () => {
  const actual = await vi.importActual('react-router');
  return {
    ...actual,
    useParams: vi.fn(() => ({
      podcastId: podcastListMock.feed.entry[0].id.attributes['im:id'],
    })),
  };
});

describe('PodcastDetail', () => {
  it('should render successfully', () => {
    const { baseElement, container } = render(
      <BrowserRouter>
        <PodcastDetailPage />
      </BrowserRouter>
    );
    expect(baseElement).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('should render the podcasts bar', () => {
    render(
      <BrowserRouter>
        <PodcastDetailPage />
      </BrowserRouter>
    );

    expect(screen.queryByTestId('podcasts_bar')).toBeTruthy();
  });

  it('should render the podcasts table', () => {
    render(
      <BrowserRouter>
        <PodcastDetailPage />
      </BrowserRouter>
    );

    expect(screen.queryByTestId('podcasts_table')).toBeTruthy();
  });
});
