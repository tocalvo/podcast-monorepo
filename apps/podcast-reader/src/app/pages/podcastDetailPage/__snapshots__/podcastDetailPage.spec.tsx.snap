// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PodcastDetail > should render successfully 1`] = `
<div>
  <div
    class="_container_c11367"
  >
    <div
      class="_card_152e7e"
      data-testid="podcasts_bar"
    >
      <a
        class="_image-link_152e7e"
        data-discover="true"
        href="/podcast/1535809341"
      >
        <img
          alt=""
          class="_image_152e7e"
          src="https://is1-ssl.mzstatic.com/image/thumb/Podcasts113/v4/f2/21/fa/f221fabd-017f-5125-633b-f1fe4f39802a/mza_182995249085044287.jpg/55x55bb.png"
        />
      </a>
      <hr
        class="_line_152e7e"
      />
      <a
        data-discover="true"
        href="/podcast/1535809341"
      >
        <p
          class="_title_152e7e"
        >
          The Joe Budden Podcast
        </p>
      </a>
      <a
        data-discover="true"
        href="/podcast/1535809341"
      >
        <p
          class="_italic_152e7e"
        >
          by 
          The Joe Budden Network
        </p>
      </a>
      <hr
        class="_line_152e7e"
      />
      <p
        class="_title2_152e7e"
      >
        Description:
      </p>
      <p
        class="_italic_152e7e"
      >
        Tune into Joe Budden and his friends. Follow along the crazy adventures of these very random friends.
      </p>
    </div>
    <div
      class="_episodes-container_b3522f"
      data-testid="podcasts_table"
    >
      <div
        class="_card_b3522f"
      >
        <p>
          <b>
            Episodes: 
            2
          </b>
        </p>
      </div>
      <div
        class="_card_b3522f _striped-table_b3522f"
      >
        <table>
          <thead>
            <tr>
              <th>
                Title
              </th>
              <th>
                Date
              </th>
              <th>
                Duration
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <a
                  class="_link_b3522f"
                  data-discover="true"
                  href="/podcast/1535809341/episode/1000677977070"
                >
                  Episode 778 | "Bottom Of The 9th"
                </a>
              </td>
              <td>
                08/07/1987
              </td>
              <td>
                12:00
              </td>
            </tr>
            <tr>
              <td>
                <a
                  class="_link_b3522f"
                  data-discover="true"
                  href="/podcast/1535809341/episode/1000677608756"
                >
                  Episode 777 | "The Spectrum"
                </a>
              </td>
              <td>
                08/07/1987
              </td>
              <td>
                12:00
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
